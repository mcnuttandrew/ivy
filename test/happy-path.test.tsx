import React from 'react';
import {mount} from 'enzyme';
import AppWrap from '../src/components/app-wrap';

test('Happy path', done => {
  const $ = mount(<AppWrap />);
  // should find the home page rendered as expected
  expect($.find('.home-container-contents').text()).toMatchSnapshot();
  // should find the expected number of default templates
  expect($.find('.home-template-row').length).toBe(9);
  // click on the polestar template
  $.find('.Ivy-Authors-Polestar a')
    .first()
    .simulate('click', {button: 0});

  // setTimeout(() => {
  // console.log($.text());
  // the text of the encoding column should render as expected
  expect($.find('.encoding-column').text()).toMatchSnapshot();
  // the text of the data column should render as expected
  expect($.find('.column.background-2').text()).toMatchSnapshot();
  // the chart area should hvae the default charts loaded
  expect($.find('.chart-container').text()).toMatchSnapshot();
  done();
  // }, 5000);
});
