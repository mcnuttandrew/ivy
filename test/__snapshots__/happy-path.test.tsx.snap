// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Happy path 1`] = `
"In order visualize your data, you need to pick a template to work in. To begin either select a template from the selection in main pane (to the right), or use this panel to search. Search can either happen through text:


Search for template


 or by search for templates that match the data you are interested in visualizing, which you can do using this widget


Data target searchdrop a field hereClearSelect a valueAcceleration 123Cylinders 123Displacement 123Horsepower 123Miles_per_Gallon 123Weight_in_lbs 123Name ABCOrigin ABCYear ðŸ•‘


 You can also facilitate browsing by sorting: 


Sorttemplate namemin required measuresmin required dimensionsmin required timesmax allowed fieldscomplexityReverse SortSearch ShortcutsUnvariatesExplorationReset"
`;

exports[`Happy path 2`] = `"Data cars.jsonCHANGEdimension (2)hideNameOriginmeasure (6)hideAccelerationCylindersDisplacementHorsepowerMiles_per_GallonWeight_in_lbstime (1)hideYear Filter drop a field here"`;

exports[`Happy path 3`] = `"No templates match your query"`;
